src/rov/rov_led_controller/rov_led_controller/led_controller.py

```diff
@@ -1,129 +1,200 @@
         # Listen for requests from the UI
         self.ui_subscriber = self.create_subscription(String, 'ui_requests', self.ui_request_callback, 10)
 
-        PIXELS.fill(BLACK)
+        PIXELS.LED_OFF_ALL()
+        
 
     def ui_request_callback(self, msg):
-        if msg.data == 'get_animations':
+        # Split message
+        split = msg.data.split(",")
+        msgs = [s.strip() for s in split]
+
+        # Change brightness
+        if len(msgs) == 2:
+            try:
+                brightness = float(msgs[1])
+                if brightness > 1.0:
+                    brightness = 1
+                elif brightness < 0:
+                    brightness = 0
+                PIXELS.set_brightness(brightness)
+                self.get_logger().info("Changed brightness to: %f" % brightness)
+            except:
+                self.get_logger().info("Second argument, brightness, should be a float.")
+        
+        # Change animation and colors
+        if msgs[0] == 'get_animations':
             # Send the list of preprogrammed animations to the UI
             animations_msg = String()
             animations_msg.data = "\n".join(animation for animation in self.animations)
             self.publisher_.publish(animations_msg)
             self.get_logger().info('Sent preprogrammed animations to UI')
-        elif msg.data == 'get_colors':
+        elif msgs[0] == 'get_colors':
             colors_msg = String()
             colors_msg.data = "\n".join(color for color in self.colors)
             self.publisher_.publish(colors_msg)
             self.get_logger().info('Sent preprogrammed colors to UI')
         else:
             # Check if the requested animation exists
-            if msg.data in self.animations:
+            if msgs[0] in self.animations:
                 # Call the method corresponding to the requested animation
-                
-                if ledState and (lastAnimation != msg.data):
-                    exec(f"self.{msg.data}({ledColor}, 0.1)")()
-                else:
-                    PIXELS.brightness(0)
-                lastAnimation = msg.data
-            elif msg.data in self.colors:
-                lastColor = ledColor
-                if ledState and (lastColor != msg.data):
-                    exec(f"self.{lastAnimation}({msg.data}, 0.1)")()
-
-
-    def on_set_rov_gpio(self, message: PinState):
-        if message.pin not in LED_PIN:
-            pass
-        ledState = message.state
+                self.currentAnimation = msgs[0]
+                self.get_logger().info('New animation received: %s' % msgs[0])
+                self.animations[msgs[0]](self.colors[self.ledColor])
+            elif msgs[0] in self.colors:
+                # Change the color if message changes colors
+                self.ledColor = msgs[0]
+                self.get_logger().info('New color received: %s' % msgs[0])
+                self.animations[self.currentAnimation](self.colors[msgs[0]])
+            
```

Comment:
Add a last case here which handles if it's "none of the above", aka an invalid message. Print out a something like "oi butt face you entered [blank] and that's wrong" - feel free to quote me